src/libs/OptionsListUtils.js

```diff
@@ -209,26 +209,35 @@ function getSearchText(report, reportName, personalDetailList, isChatRoomOrPolic
 }
 
 /**
- * If the report or the report actions have errors, return
- * CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR, otherwise an empty string.
- *
+ * Get an object of error messages keyed by microtime by combining all error objects related to the report.
  * @param {Object} report
  * @param {Object} reportActions
- * @returns {String}
+ * @returns {Object}
  */
-function getBrickRoadIndicatorStatusForReport(report, reportActions) {
+function getAllReportErrors(report, reportActions) {
     const reportErrors = report.errors || {};
     const reportErrorFields = report.errorFields || {};
     const reportID = report.reportID;
     const reportsActions = reportActions[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`] || {};
+    const reportActionErrors = {};
+    _.chain(reportsActions)
+        .filter(action => !_.isEmpty(action.errors))
+        .each(action => _.extend(reportActionErrors, action.errors));
+
+    // All error objects related to the report. Each value in the sources contains error messages keyed by microtime
+    const errorSources = {
+        reportErrors,
+        ...reportErrorFields,
+        reportActionErrors,
+    };
 
-    const hasReportFieldErrors = _.some(reportErrorFields, fieldErrors => !_.isEmpty(fieldErrors));
-    const hasReportActionErrors = _.some(reportsActions, action => !_.isEmpty(action.errors));
+    // Combine all error messages keyed by microtime into one object
+    const allReportErrors = {};
+    _.chain(errorSources)
+        .filter(errors => !_.isEmpty(errors))
+        .each(errors => _.extend(allReportErrors, errors));
```

Comment:
Is this really doing a reduce? I think so, right? Wouldn't it be better to use the reduce method to make code clearer?

Reply:
Oh yeah good point. I'm on it.